index.js

```diff
@@ -81,45 +82,98 @@ task3 = function() {
 			blue: blue,
 		};
 	}
-	var getColorInBetween = function(color1, color2){
+	var getColorInBetween = function(color1, color2,percent){
 		var getMiddleNumber = function(num1, num2, percent){
 			var ratio = percent/100;
 			return Math.floor(num1*ratio + num2*(1-ratio));
 		}
-		var red = getMiddleNumber(color1.red, color2.red, 50);
-		var green = getMiddleNumber(color1.green, color2.green, 50);
-		var blue = getMiddleNumber(color1.blue, color2.blue, 50);
+		var red = getMiddleNumber(color1.red, color2.red, percent);
+		var green = getMiddleNumber(color1.green, color2.green, percent);
+		var blue = getMiddleNumber(color1.blue, color2.blue, percent);
 		return getColor(red, green, blue);
 	}
 
 	var color1 = getColor(250,100,200);
 	var color2 = getColor(200,250,250);
-	var color3 = getColorInBetween(color1,color2);
-	console.log(color3); // See the output in JavaScript Console.
+	var color3 = getColorInBetween(color1,color2,20);
+	console.log(color3);
+
+  // See the output in JavaScript Console.
 	// Notice that we are creating 2 colors, and then finding a color in between them.
 	// TODO: Add an argument called percent to getColorInBetween() so that we don't
 	// have to get the 'center' color in between color1 and color2, and can get anything.
 };
 
 // TODO: Task four, combine it all, all by yourself.
 
-var task4 = function() {
+var task4 = function() 
+{
 	// Here, you will write/copy from above code to achieve the following:
 	// Create a timer triggered by the <enter> key that changes the background
 	// color of the page with each second, except, here's the deal:
 	// The color goes from color1 slowly to color2, as the time passes.
 	// Example: black, dark grey, medium gray, light grey, white.
 
 	// Some sample code could look like:
+	var getColorString = function(color) 
+	{
+		return "rgb(" + color.red + "," + color.green + "," + color.blue + ")";
+	}
+	var getColor = function(red, green, blue)
+	{
+		return {
+			red: red,
+			green: green,
+			blue: blue,
+		};
+	}
+	var Percent=100;
 	var startColor = getColor(100,200,250);
 	var endColor = getColor(200,250,100);
 	var body = document.getElementsByTagName('body')[0];
-	body.addEventListener("keyup", function(keyevent) {
+	body.addEventListener("keyup", function(keyevent) 
+	{
 	    if (keyevent.keyCode == 13) { // enter key keycode
 	        startTimer(5);
 	        return false;
 	    }
 	});
+	var timedisplay = document.getElementById('timedisplay');
+	var startTimer = function(seconds)
+	{
+		var interval = setInterval(function()
+		{
+			seconds--;
+			timedisplay.innerHTML=seconds;
+			body.style.background = color();
+			Percent-=20;
+			if (seconds < 0)
+			{
+				clearInterval(interval); // delete the timer
+				timedisplay.innerHTML="Hurray!"; //display finishing message
+			}
+		}, 1000);
+	};
+	var color= function ()
+	{
+	var getColorInBetween = function(color1,color2,percent)
```

Comment:
Block doesn't have indentation. Try to indent properly so that the reviewer can easily read the code.